src/NodeApi/DotNetHost/NativeHost.cs

```diff
@@ -43,28 +40,19 @@ public static napi_value InitializeModule(napi_env env, napi_value exports)
     {
         Trace($"> NativeHost.InitializeModule({env.Handle:X8}, {exports.Handle:X8})");
 
-        using var scope = new JSValueScope(JSValueScopeType.RootNoContext, env);
-
-        ResolveImports();
         try
         {
-            var host = new NativeHost();
-
-            // Define a dispose method implemented by the native host that closes the CLR context.
-            // The managed host proxy will pass through dispoose calls to this callback.
-            new JSValue(exports, scope).DefineProperties(new JSPropertyDescriptor(
-                "dispose", (_) => { host.Dispose(); return default; }));
-
-            try
-            {
-                exports = host.InitializeManagedHost(env, exports);
-            }
-            catch
-            {
-                host.Dispose();
-                throw;
-            }
+            using JSValueScope scope = new(JSValueScopeType.RootNoContext, env);
```

Comment:
> using JSValueScope scop [](http://example.com/codeflow?start=11&length=24)

Why it was moved inside of try block? Does it mean that the code in the catch block will now always crash the app instead of reporting errors? The `JSValue.CreateError` requires a `JSValueScope`.

Reply:
You're right. I was moving things around here to troubleshoot the module context issue, but the scope definitely should be outside the try block. Fixed.